src/experimental/collection_new_iterable_model/shared/collectionInternalCommunication/server/TreeSetServerDispatcher.hpp

```diff
@@ -0,0 +1,57 @@
+                        typename __ReverseConstIteratorType,                                                                                                    // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __cbeginFunction )            ()                                      const -> __ConstIteratorType,           // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __cendFunction )              ()                                      const -> __ConstIteratorType,           // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __crbeginFunction )           ()                                      const -> __ReverseConstIteratorType,    // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __crendFunction )             ()                                      const -> __ReverseConstIteratorType,    // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __newAddressFunction )        ( __ElementType const *, bool * )             -> __ElementType *,               // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __removeConstFunction )       ( __AbstractConstIteratorType const & )       -> bool,                          // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __removeConstArrayFunction )  ( __AbstractConstIteratorType const * const *, Size )       -> Size,                          // NOLINT(bugprone-reserved-identifier)
+                        auto ( __ServiceType :: * __findConstFunction )         ( __ElementType const & )                const -> __ConstIteratorType           // NOLINT(bugprone-reserved-identifier)
+                > class __TreeSetServerDispatcher :     // NOLINT(bugprone-reserved-identifier)
+                        public __SetServerDispatcher <
+                                __ServerType,
+                                __ServiceType,
+                                __ElementType,
+                                __AbstractConstIteratorType,
+                                __ConstIteratorType,
+                                __cbeginFunction,
+                                __cendFunction,
+                                __newAddressFunction,
+                                __removeConstFunction,
+                                __removeConstArrayFunction,
+                                __findConstFunction
+                        > {
+
+                        protected:
+
+                            auto __crbeginLocal () const noexcept -> __ReverseConstIteratorType;        // NOLINT(bugprone-reserved-identifier)
+
+                            auto __crendLocal () const noexcept -> __ReverseConstIteratorType;          // NOLINT(bugprone-reserved-identifier)
```

Comment:
mark each function with the respective access modifier

protected:
    fun1 

protected:
    fun2